src/shared/components/home/signup.tsx

```diff
@@ -71,9 +71,13 @@ export class Signup extends Component<any, State> {
 
   async fetchCaptcha() {
     this.setState({ captchaRes: LOADING_REQUEST });
+    HttpService.client.setHeaders({
```

Comment:
It would be helpful to add functionality to the JS client to set headers for individual requests. This time it doesn't matter too much: the signup page should only be visible to unauthenticated users, so we don't have to worry about overwriting the authentication header. However, I can see this making things needlessly difficult down the line.

Reply:
If its just a few times too, it might be useful to just create either a separate client without headers on demand, or in HTTPService. Adding optional function params to every lemmy-js-client call might be overkill.